src/module/data/pseudo-documents/pseudo-document.mjs

```diff
@@ -100,13 +100,8 @@ export default class PseudoDocument extends foundry.abstract.DataModel {
    * @type {string}
    */
   get fieldPath() {
-    const fp = this.schema.fieldPath;
-    let path = fp.slice(0, fp.lastIndexOf("element") - 1);
-
-    if (this.parent instanceof PseudoDocument) {
-      path = [this.parent.fieldPath, this.parent.id, path].join(".");
-    }
-
+    let path = this.parent.constructor.metadata.embedded[this.documentName];
+    if (this.parent instanceof PseudoDocument) path = [this.parent.fieldPath, this.parent.id, path].join(".");
```

Comment:
I found a bug here though I did not test whether it existed on the main branches. It prevented updates, but regardless the new edition is simpler.